python/lsst/phosim/utils/phosim_repackager.py

```diff
@@ -97,110 +101,111 @@ def repackage(self, phosim_amp_files, out_dir='.'):
-        # relative to Camera coordinates) from which obs_obs_lsst can
-        # infer the CCD-wide WCS.
-        sensor[0].header['RATEL'] = sensor[1].header['RA_DEG']
-        sensor[0].header['DECTEL'] = sensor[1].header['DEC_DEG']
-        # The rotation angle has different names in different versions
-        try:
-            sensor[0].header['ROTANGLE'] = sensor[1].header['ROTANGZ']
-        except KeyError:
-            try:
-                sensor[0].header['ROTANGLE'] = sensor[1].header['ROTANG']
-            except KeyError as e:
-                raise(e)
 
-        outfile = self.mef_filename(phosim_amp_files[0], out_dir=out_dir)
-        sensor.writeto(outfile, overwrite=True)
+        serial = detectors.getSerial()  # eg. ITL-4400B-029
+        CCD_MANU, CCD_TYPE, CCD_NUM = serial.split('-')
+        sensor[0].header['LSST_NUM'] = serial
+        sensor[0].header['CCD_MANU'] = CCD_MANU  # eg. ITL
+        sensor[0].header['CCD_TYPE'] = CCD_TYPE  # eg. 4400B
 
-    @staticmethod
-    def mef_filename(phosim_amp_file, out_dir='.'):
-        """
-        Construct the filename of the output MEF file based
-        on a phosim single amplifier filename.
-        """
-        tokens = os.path.basename(phosim_amp_file).split('_')
-        # Remove the channel identifier.
-        outfile = '_'.join(tokens[:6] + tokens[7:])
-        outfile = os.path.join(out_dir, outfile)
-        # astropy's gzip compression is very slow, so remove any .gz
-        # extension from the computed output filename and gzip the
-        # files later.
-        if outfile.endswith('.gz'):
-            outfile = outfile[:-len('.gz')]
-        return outfile
+        # NB: I get [1:4072,1:2000], whereas  BOT has '[1:4072,1:4000]'
+        sensor[0].header['DETSIZE'] = noao_section_keyword(detectors.getBBox())
+        sensor[0].header['INSTRUME'] = 'lsstCam'
+        sensor[0].header['TELESCOP'] = 'LSST'
+
+        sensor[0].header['TELCODE'] = TELCODE
+        sensor[0].header['CONTRLLR'] = CONTRLLR
+        # Set sequence number from OBSID, eg. 9006001, taking
+        # the last 6 digits
+        SEQNUM = int(sensor[0].header['RUNNUM'][-6:])
+        sensor[0].header['SEQNUM'] = SEQNUM
+
+        OBSID = "%s_%s_%s_%s"%(TELCODE, CONTRLLR, DAYOBS, str(SEQNUM).zfill(6))
+        sensor[0].header['OBSID'] = OBSID
+
+        sensor[0].header['TESTTYPE'] = 'PHOSIM'
+        sensor[0].header['IMGTYPE'] = 'SKYEXP'
+        sensor[0].header['RAFTBAY'] = raft
+        sensor[0].header['CCDSLOT'] = ccdslot
+        sensor[0].header['RASTART'] = sensor[1].header['RA_DEG']
+        sensor[0].header['DECSTART'] = sensor[1].header['DEC_DEG']
+        sensor[0].header['ROTSTART'] = sensor[1].header['ROTANG']
+        sensor[0].header['RA'] = sensor[1].header['RA_DEG']
+        sensor[0].header['DEC'] = sensor[1].header['DEC_DEG']
+        sensor[0].header['ROTPA'] = sensor[1].header['ROTANG']
+        sensor[0].header['ROTPOS'] = sensor[1].header['ROTANG']
```

Comment:
Does the `ROTPA` equal to `ROTPOS`?

Reply:
For header conformity with the lsstCam translator.